src/scrot_selection.c

```diff
@@ -232,15 +232,22 @@ static bool scrotSelectionGetUserSel(struct SelectionRect *selectionRect)
                 scrotSelectionMotionDraw(rx, ry, ev.xmotion.x, ev.xmotion.y);
             break;
         case ButtonPress:
-            isButtonPressed = true;
-            rx = ev.xbutton.x;
-            ry = ev.xbutton.y;
-            target = scrotGetWindow(disp, ev.xbutton.subwindow, ev.xbutton.x, ev.xbutton.y);
-            if (target == None)
-                target = root;
+            if (ev.xbutton.button == 1) {
+                isButtonPressed = true;
+                rx = ev.xbutton.x;
+                ry = ev.xbutton.y;
+                target = scrotGetWindow(disp, ev.xbutton.subwindow, ev.xbutton.x, ev.xbutton.y);
+                if (target == None)
+                    target = root;
+            }
             break;
         case ButtonRelease:
-            done = DONE;
+            if (ev.xbutton.button == 1)
+                done = DONE;
+            else if (ev.xbutton.button == 3) {
+                warnx("Right mouse button was pressed, aborting shot");
```

Comment:
It says "pressed" but this is operating on release, not press.

Reply:
Looks like it doesn't generate a button press event if one of the other button is held down?

```console
[scrot master]~> git diff
diff --git a/src/scrot_selection.c b/src/scrot_selection.c
index 9483e9c..8aaf666 100644
--- a/src/scrot_selection.c
+++ b/src/scrot_selection.c
@@ -232,6 +232,7 @@ static bool scrotSelectionGetUserSel(struct SelectionRect *selectionRect)
                 scrotSelectionMotionDraw(rx, ry, ev.xmotion.x, ev.xmotion.y);
             break;
         case ButtonPress:
+            warnx("got ButtonPress (%u)", ev.xbutton.button);
             isButtonPressed = true;
             rx = ev.xbutton.x;
             ry = ev.xbutton.y;
@@ -240,6 +241,7 @@ static bool scrotSelectionGetUserSel(struct SelectionRect *selectionRect)
                 target = root;
             break;
         case ButtonRelease:
+            warnx("got ButtonRelease (%u)", ev.xbutton.button);
             done = DONE;
             break;
         case KeyPress:
[scrot master]~> ./scrot -s - >/dev/null
scrot: got ButtonPress (1)
scrot: got ButtonRelease (3)
```

I was expecting to see `press 1, press 3, release 3` but the `press 3` event didn't come...

Reply:
While the warning does say "press", actually trying to match a press event might cause unwanted behavior.
The reason you're not seeing `press 3` is because the `scrotSelectionMotionDraw` function applies the following event mask:
```c
static void scrotSelectionMotionDraw(int x0, int y0, int x1, int y1)
{
    const struct Selection *const sel = &selection;
    const unsigned int EVENT_MASK = ButtonMotionMask | ButtonReleaseMask;
    ...
}
```
So once you start moving the mouse after starting the selection, clicking any other mouse buttons won't actually do anything (note that if you hold down the left mouse button, without moving the mouse, press events for other mouse buttons will get picked up).
The "fix" for this is adding the `ButtonPressMask` flag, but like I said, this might cause unwanted behavior. I'm not familiar enough with the codebase to know for sure.

Practically, it doesn't matter if what actually triggers the abort is a button press or release. If anything, making it trigger on a release event makes it more user friendly, because the user might want to cancel, but then while holding down the right mouse button decide to take the screenshot anyway.

I suppose a better warning message would be "Right mouse button was *clicked*, aborting"?